setup/src/Magento/Setup/Console/Command/AdminUserCreateCommand.php

```diff
@@ -55,15 +55,16 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $errors = $this->validate($input);
+        $data = $this->getUserData($input, $output);
+        $errors = $this->validate($data);
         if ($errors) {
             $output->writeln('<error>' . implode('</error>' . PHP_EOL .  '<error>', $errors) . '</error>');
             return;
         }
         $installer = $this->installerFactory->create(new ConsoleLogger($output));
-        $installer->installAdminUser($input->getOptions());
+        $installer->installAdminUser($data);
         $output->writeln(
-            '<info>Created Magento administrator user named ' . $input->getOption(AdminAccount::KEY_USER) . '</info>'
+            '<info>Created Magento administrator user named ' . $data[AdminAccount::KEY_USER] . '</info>'
```

Comment:
I'm not totally against this approach in this particular case, so I'm ok, if it will be as is.
Just, in general, I'd prefer to not rely on internal logic of a method, but rather rely on the interface/signature. Even, if it is in the same class. Sometimes, it may happen that different developers touch the code and they may miss usage of data returned by a method.
Also, in this case, the method is protected, which means that it's behavior may be modified (close to 0 chance in this particular case, but I'm saying in general).

Again, it's ok to leave it as is in this case.